### PR TITLE
7 integration spec view

### PR DESCRIPTION
**Pull request by Ukaypromise**
### Hello Dear Reviewer 👋🏾 
**For this PR I have been able to implement the following requirements:**

- [x] Use Capybara to write integration tests for each view in your project.
- [x] Write unit test for User index page.
- [x] Write unit test for user show page:
- [x] Add unit test for User post index page.
- [x] Add unit test for Post show page

**You can gracefully take your time and review this PR and please leave comments where i need to improve. Thank you.**

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    it 'displays the user name' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content('Promise')
+    end
+
+    it 'displays total posts by the user has written' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content('2')
+    end
+
```
Comment (tufoinnkuo10) [inline spec/views/post_index_spec.rb:56]: - [x] great job. kindly add  Tests that asserts you can see a post's title

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    it 'can see some of the post body' do
+      visit user_posts_path(@user1.id)
+      expect(page).to have_content('First post')
+      expect(page).to have_content('Second post')
+      expect(page).to have_content('This is my first post')
+      expect(page).to have_content('This is my second post')
+    end
+
```
Comment (tufoinnkuo10) [inline spec/views/post_index_spec.rb:64]: - [ ] please kinldy add test for that asserts you can see the first comments on a post.

File: spec/views/post_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    it 'displays total comments and likes' do
+      visit user_post_path(@user1, @post1)
+      expect(page).to have_content('2')
+    end
+
+    it 'displays total comments and likes' do
+      visit user_post_path(@user1, @post2)
+      expect(page).to have_content('1')
+    end
```
Comment (tufoinnkuo10) [inline spec/views/post_show_spec.rb]: - [x] both test cases have the same name. please use different names for the respective test cases

File: spec/views/user_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    visit user_path(@user)
+    expect(page).to have_content('Fullstack Developer')
+  end
+
+  it 'displays the right Number of posts' do
+    visit user_path(@user)
+    expect(page).to have_content('2')
+  end
+
+  it 'displays all the available users posts' do
```
Comment (tufoinnkuo10) [inline spec/views/user_show_spec.rb]: - [x] great job. Please add Tests that asserts you can see the user's first 3 posts and not all posts

File: spec/views/user_show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    click_link 'see all posts'
+    expect(page).to have_current_path(user_posts_path(@user))
+  end
+
+  it 'click a users post, it redirects to the posts show page' do
+    visit user_path(@user)
+    click_link 'First post'
+    expect(page).to have_current_path(user_post_path(@user, @post1))
+  end
+
```
Comment (tufoinnkuo10) [inline spec/views/user_show_spec.rb:73]: - [x] kindly add a Tests that redirects to the user's post's index page when you click on a **see all posts**.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Ukaypromise 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally


Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/views/post_index_spec.rb (hunk 1 above)
Comment (Ukaypromise) [reply spec/views/post_index_spec.rb:56]: Implemented this !

File: spec/views/post_index_spec.rb (hunk 2 above)
Comment (Ukaypromise) [reply spec/views/post_index_spec.rb:64]: Fixed

File: spec/views/post_show_spec.rb (hunk 3 above)
Comment (Ukaypromise) [reply spec/views/post_show_spec.rb]: Done!

File: spec/views/user_show_spec.rb (hunk 4 above)
Comment (Ukaypromise) [reply spec/views/user_show_spec.rb]: Done!

File: spec/views/user_show_spec.rb (hunk 5 above)
Comment (Ukaypromise) [reply spec/views/user_show_spec.rb:73]: Done

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @Ukaypromise      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 




 

- [x] No Linter Errors







Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**